acceptance/aws/stack_test.go

```diff
@@ -63,4 +66,81 @@ func TestStack(t *testing.T) {
+  Test1: 1
+  Test2: 2
+  Test3: 3
+Tags:
+  Tag1: 1
+  Tag2: 2
+  Tag3: 3
+StackPolicy:
+  Statement:
+  - Effect: Allow
+    Resource: "*"
+`, "yml")
+
+	defer stackFileYMLClean()
+
+	t.Run("update", func(t *testing.T) {
+		Template("update stack name=some-stack template-file="+tplFilePath+" stack-file="+stackFileYMLPath+" parameters=Test1:a,Test2:b tags=Tag1:a,Tag2:b policy-file="+polFilePath).Mock(&cloudformationMock{
+			UpdateStackFunc: func(input *cloudformation.UpdateStackInput) (*cloudformation.UpdateStackOutput, error) {
+				return &cloudformation.UpdateStackOutput{StackId: String("any-stack-id")}, nil
+			}}).ExpectInput("UpdateStack", &cloudformation.UpdateStackInput{
+			StackName:       String("some-stack"),
+			TemplateBody:    String("tpl body content"),
+			Parameters:      []*cloudformation.Parameter{{ParameterKey: String("Test1"), ParameterValue: String("a")}, {ParameterKey: String("Test2"), ParameterValue: String("b")}, {ParameterKey: String("Test3"), ParameterValue: String("3")}},
+			Tags:            []*cloudformation.Tag{{Key: String("Tag1"), Value: String("a")}, {Key: String("Tag2"), Value: String("b")}, {Key: String("Tag3"), Value: String("3")}},
+			StackPolicyBody: String("policy content"),
+		}).ExpectCalls("UpdateStack").Run(t)
+	})
+
+	_, stackFileJSONPath, stackFileJSONClean := generateTmpFileWithExt(`
+		{"Parameters":{"Test1":"1","Test2":"2","Test3":"3"},"Tags":{"Tag1":"1","Tag2":"2","Tag3":"3"},"StackPolicy":{"Statement":[{"Effect":"Allow","Resource":"*"}]}}`, "json")
+
+	defer stackFileJSONClean()
+
+	t.Run("update", func(t *testing.T) {
+		Template("update stack name=some-stack template-file="+tplFilePath+" stack-file="+stackFileJSONPath+" parameters=Test1:a,Test2:b tags=Tag1:a,Tag2:b").Mock(&cloudformationMock{
+			UpdateStackFunc: func(input *cloudformation.UpdateStackInput) (*cloudformation.UpdateStackOutput, error) {
+				return &cloudformation.UpdateStackOutput{StackId: String("any-stack-id")}, nil
+			}}).ExpectInput("UpdateStack", &cloudformation.UpdateStackInput{
+			StackName:       String("some-stack"),
+			TemplateBody:    String("tpl body content"),
+			Parameters:      []*cloudformation.Parameter{{ParameterKey: String("Test1"), ParameterValue: String("a")}, {ParameterKey: String("Test2"), ParameterValue: String("b")}, {ParameterKey: String("Test3"), ParameterValue: String("3")}},
+			Tags:            []*cloudformation.Tag{{Key: String("Tag1"), Value: String("a")}, {Key: String("Tag2"), Value: String("b")}, {Key: String("Tag3"), Value: String("3")}},
+			StackPolicyBody: String(`{"Statement":[{"Effect":"Allow","Resource":"*"}]}`),
+		}).ExpectCalls("UpdateStack").Run(t)
+	})
+
+}
+
+func generateTmpFileWithExt(content, ext string) (*os.File, string, func()) {
+	file, err := ioutil.TempFile("", "awless-at-tmpfile")
+	if err != nil {
+		panic(err)
+	}
+
+	newFilePath := strings.Join([]string{file.Name(), ext}, ".")
+	if err = os.Rename(file.Name(), newFilePath); err != nil {
+		panic(err)
+	}
```

Comment:
Shall we instead of a extra rename use just https://golang.org/pkg/io/ioutil/#TempDir, Join ioutil.WriteFile?

Reply:
Sure, make sense.